packages/core/src/list-transactions.ts

```diff
@@ -48,86 +48,94 @@ export async function listTransactions(
+  for await (const message of deploymentLoader.readFromJournal()) {
+    if (message.type !== JournalMessageType.TRANSACTION_SEND) {
       continue;
     }
 
-    for (const networkInteraction of exState.networkInteractions) {
-      assertIgnitionInvariant(
-        networkInteraction.type === "ONCHAIN_INTERACTION",
-        "Expected network interaction to be an onchain interaction"
-      );
-
-      for (const [
-        index,
-        transaction,
-      ] of networkInteraction.transactions.entries()) {
-        switch (exState.type) {
-          case ExecutionSateType.DEPLOYMENT_EXECUTION_STATE: {
-            transactions.push({
-              type: exState.type,
-              from: exState.from,
-              txHash: transaction.hash,
-              status: getTransactionStatus(
-                transaction,
-                index === networkInteraction.transactions.length - 1
-              ),
-              name: exState.contractName,
-              address:
-                transaction.receipt?.status === TransactionReceiptStatus.SUCCESS
-                  ? exState.result?.type === ExecutionResultType.SUCCESS
-                    ? exState.result.address
-                    : undefined
-                  : undefined,
-              params: exState.constructorArgs,
-              value: networkInteraction.value,
-            });
-
-            break;
-          }
-          case ExecutionSateType.CALL_EXECUTION_STATE: {
-            const artifact = await deploymentLoader.loadArtifact(
-              exState.artifactId
-            );
-
-            transactions.push({
-              type: exState.type,
-              from: exState.from,
-              txHash: transaction.hash,
-              status: getTransactionStatus(
-                transaction,
-                index === networkInteraction.transactions.length - 1
-              ),
-              name: `${artifact.contractName}#${exState.functionName}`,
-              to: networkInteraction.to,
-              params: exState.args,
-              value: networkInteraction.value,
-            });
-
-            break;
-          }
-          case ExecutionSateType.SEND_DATA_EXECUTION_STATE: {
-            transactions.push({
-              type: exState.type,
-              from: exState.from,
-              txHash: transaction.hash,
-              status: getTransactionStatus(
-                transaction,
-                index === networkInteraction.transactions.length - 1
-              ),
-              to: networkInteraction.to,
-              value: networkInteraction.value,
-            });
-
-            break;
-          }
-        }
+    const exState = deploymentState.executionStates[message.futureId];
+
+    assertIgnitionInvariant(
+      doesSendTransactions(exState),
+      "Expected execution state to be a type that sends transactions"
+    );
+
+    const networkInteraction =
+      exState.networkInteractions[message.networkInteractionId - 1];
```

Comment:
This `-1` caught me by surprise tbh. Do we always have the network interaction ids be 1, 2, 3, ...

TBH I don't remember, and I couldn't find it easily in the code.

Reply:
I search the code, and we normally do

```ts
const onchainInteraction = executionState.networkInteractions.find(
    (interaction) => interaction.id === networkInteractionId
  );
```

Reply:
The `-1` is because i'm indexing it directly, not using a `find`. As far as I know, we've always started the networkInteraction ID's at 1